apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -1,65 +1,109 @@
+    private createNotification: CreateNotification,
     private processSubscriber: ProcessSubscriber,
     private jobRepository: JobRepository,
-    private addJobUsecase: AddJob,
     private notificationRepository: NotificationRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
     protected createExecutionDetails: CreateExecutionDetails
   ) {}
 
   async execute(command: TriggerEventCommand) {
-    await this.validateTransactionIdProperty(command.transactionId, command.organizationId, command.environmentId);
+    const { actor, environmentId, identifier, organizationId, to, userId } = command;
 
-    const template = await this.notificationTemplateRepository.findByTriggerIdentifier(
-      command.environmentId,
-      command.identifier
-    );
+    await this.validateTransactionIdProperty(command.transactionId, organizationId, environmentId);
 
     Sentry.addBreadcrumb({
       message: 'Sending trigger',
       data: {
-        triggerIdentifier: command.identifier,
+        triggerIdentifier: identifier,
       },
     });
 
+    const template = await this.notificationTemplateRepository.findByTriggerIdentifier(
+      command.environmentId,
+      command.identifier
+    );
+
+    /*
+     * Makes no sense to execute anything if template doesn't exist
+     * TODO: Send a 404?
+     */
+    if (!template) {
+      const message = 'Notification template could not be found';
+      Logger.error(message, LOG_CONTEXT);
+      throw new ApiException(message);
+    }
+
     const jobs: Omit<JobEntity, '_id'>[][] = [];
 
-    for (const subscriberToTrigger of command.to) {
-      jobs.push(
-        await this.processSubscriber.execute(
-          ProcessSubscriberCommand.create({
-            identifier: command.identifier,
-            payload: command.payload,
-            overrides: command.overrides,
-            to: subscriberToTrigger,
-            transactionId: command.transactionId,
-            environmentId: command.environmentId,
-            organizationId: command.organizationId,
-            userId: command.organizationId,
-            templateId: template._id,
-            actor: command.actor,
-            template,
-          })
-        )
+    // We might have a single actor for every trigger so we only need to check for it once
+    let actorProcessed;
+    if (actor) {
+      actorProcessed = await this.processSubscriber.execute(
+        ProcessSubscriberCommand.create({
+          environmentId,
+          organizationId,
+          userId,
+          subscriber: actor,
+        })
+      );
+    }
+
+    for (const subscriber of to) {
+      const subscriberProcessed = await this.processSubscriber.execute(
+        ProcessSubscriberCommand.create({
+          environmentId,
+          organizationId,
+          userId,
+          subscriber,
+        })
       );
+
+      // If no subscriber makes no sense to try to create notification
+      if (subscriberProcessed) {
```

Comment:
Comment says it all. Before we would try to create a notification even if the subscriber didn't exist in the database.

Reply:
should we then return even before we enter this for loop?

Reply:
Do you mean instead to do:
```
if (!subscriberProcessed) {
  continue;
}
```
?
RIght now I am surrounding the loop code to execute with the condition so nothing is executed if no subscriber. And the query we have to do it, to double check in case data stored in the queue and database data is not consistent.

Reply:
nvm, I feel like I didn't notice that the code is surrounded within the same if...